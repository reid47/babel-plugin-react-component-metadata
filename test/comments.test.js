import * as babel from 'babel-core';
import plugin from '../src';

const transform = (example, options) => {
  const { code } = babel.transform(example, {
    plugins: [
      'syntax-class-properties',
      'syntax-jsx',
      options ? [plugin, options] : plugin
    ]
  });

  return code;
};

describe('comments on classes/functions', () => {
  test('mixed example', () => {
    const example = `
      // This is a comment for class Test1
      // with multiple lines.
      class Test1 extends React.Component {
        static propTypes = {
          name: PropTypes.string
        }

        render() {
          return null;
        }
      }

      /*
      * This is a comment
      * for class Test2.
      */
      class Test2 extends React.Component {
        render() {
          return null;
        }
      }
      Test2.propTypes = {
        name: PropTypes.string
      };

      // Comment for Test3
      function Test3() {
        return null;
      }
      Test3.propTypes = {name: PropTypes.string};

      /* Comment for Test4 */
      const Test4 = () => null;
      Test4.propTypes = {name: PropTypes.string};
    `;

    expect(transform(example)).toMatchSnapshot();
  });
});

xdescribe('comments prop types', () => {});
