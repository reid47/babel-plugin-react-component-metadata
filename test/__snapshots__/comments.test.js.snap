// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comments on classes/functions mixed example 1`] = `
"var _metadataHelpers = function () {
  function makeFuncType(type, optionsName) {
    return function (options) {
      var opts = {};
      opts[optionsName] = options;
      return Object.assign({
        type: type,
        required: false
      }, opts);
    };
  }

  var t = {};
  t.string = {
    type: \\"string\\",
    required: false
  };
  t.number = {
    type: \\"number\\",
    required: false
  };
  t.bool = {
    type: \\"bool\\",
    required: false
  };
  t.func = {
    type: \\"func\\",
    required: false
  };
  t.object = {
    type: \\"object\\",
    required: false
  };
  t.symbol = {
    type: \\"symbol\\",
    required: false
  };
  t.array = {
    type: \\"array\\",
    required: false
  };
  t.node = {
    type: \\"node\\",
    required: false
  };
  t.element = {
    type: \\"element\\",
    required: false
  };
  t.any = {
    type: \\"any\\",
    required: false
  };
  t.oneOf = makeFuncType(\\"oneOf\\", \\"possibleValues\\");
  t.instanceOf = makeFuncType(\\"instanceOf\\", \\"constructor\\");
  t.arrayOf = makeFuncType(\\"arrayOf\\", \\"elementType\\");
  t.objectOf = makeFuncType(\\"objectOf\\", \\"valueType\\");
  t.oneOfType = makeFuncType(\\"oneOfType\\", \\"possibleTypes\\");
  t.shape = makeFuncType(\\"shape\\", \\"innerTypes\\");

  t.isRequired = function (type) {
    return Object.assign({}, type, {
      required: true
    });
  };

  return t;
}();

// This is a comment for class Test1
// with multiple lines.
class Test1 extends React.Component {
  static propTypes = {
    name: PropTypes.string
  };

  render() {
    return null;
  }
}

/*
* This is a comment
* for class Test2.
*/
Test1.metadata = {
  props: {
    name: {
      type: metadataHelpers.string
    }
  },
  comments: {
    hasComments: true,
    lines: [\\"This is a comment for class Test1\\", \\"with multiple lines.\\"]
  }
};
class Test2 extends React.Component {
  render() {
    return null;
  }
}
Test2.metadata = {
  props: {
    name: {
      type: metadataHelpers.string
    }
  },
  comments: {
    hasComments: true,
    lines: [\\"* This is a comment\\", \\"* for class Test2.\\"]
  }
};
Test2.propTypes = {
  name: PropTypes.string
};

// Comment for Test3
function Test3() {
  return null;
}
Test3.metadata = {
  props: {
    name: {
      type: metadataHelpers.string
    }
  },
  comments: {
    hasComments: true,
    lines: [\\"Comment for Test3\\"]
  }
};
Test3.propTypes = { name: PropTypes.string };

/* Comment for Test4 */
const Test4 = () => null;
Test4.metadata = {
  props: {
    name: {
      type: metadataHelpers.string
    }
  },
  comments: {
    hasComments: true,
    lines: [\\"Comment for Test4\\"]
  }
};
Test4.propTypes = { name: PropTypes.string };"
`;
